pr

Rohith-Dasari opened the pull request.

wg-utkarsh-singh commented on models/user.go:
```diff
@@ -1,9 +1,16 @@
 
+type Role string
+
+const (
```
Please add enums to a separate file.

Rohith-Dasari replied:
Done: moved to constants/constants.go

wg-utkarsh-singh commented on handlers/coverage.out:
Remove this file.

Rohith-Dasari replied:
Removed the file as suggested and added it to .gitignore

wg-utkarsh-singh commented on handlers/grades_handler_test.go:
```diff
@@ -330,68 +347,99 @@ func TestHandler_GetTopThree(t *testing.T) {
 		})
-
 	}
 }
```
Can you add more negative scenarios?

Rohith-Dasari replied:
Thank you for pointing this out, made it more robust to negative values. Understood code coverage solely !=quality.

wg-utkarsh-singh commented on mocks/grade_service_mock.go:
```diff
@@ -55,10 +55,10 @@ func (mr *MockGradeServiceIMockRecorder) AddGrades(studentID, subjectID, Grade,
 	m.ctrl.T.Helper()
 	ret := m.ctrl.Call(m, "GetAverageOfClass", classID, semester)
-	ret0, _ := ret[0].(int)
+	ret0, _ := ret[0].(float64)
```
Why you have changes Avg from `int` to `float64`?

Rohith-Dasari replied:
Changed type of `averageOfClass` in `GetAverageOfClass` function  from `int` to `float64` to provide more precise values in the response, avoiding integer rounding. Hence, the change in mocks as well.

Rohith-Dasari commented:
1. Added more negative test cases and brought coverage of those functions back to 100%. 
<img width="660" height="105" alt="image" src="https://github.com/user-attachments/assets/cd750cc3-ba45-46f6-b68a-355749e40bf9" />
 
2.  Moved constants to constants/constants.go
3.  Removed coverage file with gitignore
4. Updated URI tested in postman(adhering rest principles and future scope)-
<img width="868" height="746" alt="image" src="https://github.com/user-attachments/assets/21951855-2a2b-436f-8135-2c551cf20c8d" />
<img width="799" height="387" alt="image" src="https://github.com/user-attachments/assets/cb6cd1f2-084c-4d7a-bb3f-0bf41aa3befa" />
